### PR TITLE
Added UserConnections function and structs

### PR DESCRIPTION
**Pull request by Necroforger**
Adds a method to fetch your own connections.
Only works on a user account.

To get another user's connections you would need to fetch their profile while on a user account.

### CODE REVIEW

File: structs.go [hunk 1]
```diff
@@ -112,6 +112,37 @@ type Session struct {
 	wsMutex sync.Mutex
 }
 
+// Connection is a Connection returned from the UserConnections endpoint
+type Connection struct {
```
Comment (iopred) [inline structs.go]: Connection is pretty generic, could we call this UserConnection

Comment (Necroforger) [reply structs.go]: GuildConnection and GuildAccount share the same fields, would it be possible to rename them to use the same structs

Comment (iopred) [reply structs.go]: Guh, don't really want to break the API, would be happy embedding:

type GuildIntegration struct {
  Integration
}

File: structs.go [hunk 2]
```diff
@@ -477,23 +508,7 @@ type GuildBan struct {
...
-	ID                string                   `json:"id"`
-	Name              string                   `json:"name"`
-	Type              string                   `json:"type"`
-	Enabled           bool                     `json:"enabled"`
-	Syncing           bool                     `json:"syncing"`
-	RoleID            string                   `json:"role_id"`
-	ExpireBehavior    int                      `json:"expire_behavior"`
-	ExpireGracePeriod int                      `json:"expire_grace_period"`
-	User              *User                    `json:"user"`
-	Account           *GuildIntegrationAccount `json:"account"`
```
Comment (iopred) [inline structs.go]: Guh...... We break the API here too.. I think maybe we just break the API, sorry to make you undo your change here, maybe just change Guild to use Integration and be done with it. Searching seems to show that noone is using GuildIntegration on github.

Comment (Necroforger) [reply structs.go]: Should I revert the last commit

**Pull request merged**